### PR TITLE
feat(apis): add completed state to podgroup&queue

### PR DESCRIPTION
**Pull request by shinytang6**
ref: https://github.com/volcano-sh/volcano/issues/2056

### CODE REVIEW

**Comment by shinytang6** [comment]
Pls help take a look :)
/cc @Thor-wl @william-wang @k82cn @Yikun @hwdef

File: pkg/apis/scheduling/v1beta1/types.go [hunk 1]
```diff
@@ -289,6 +292,8 @@ type QueueStatus struct {
 	Running int32 `json:"running,omitempty" protobuf:"bytes,4,opt,name=running"`
 	// The number of `Inqueue` PodGroup in this queue.
 	Inqueue int32 `json:"inqueue,omitempty" protobuf:"bytes,5,opt,name=inqueue"`
+	// The number of `Completed` PodGroup in this queue.
+	Completed int32 `json:"completed,omitempty" protobuf:"bytes,5,opt,name=completed"`
```
Comment (k82cn) [inline pkg/apis/scheduling/v1beta1/types.go:296]: seq num should be 6.

File: hack/generate-internal-groups.sh [hunk 2]
```diff
@@ -47,7 +47,7 @@ EXT_APIS_PKG="$4"
 GROUPS_WITH_VERSIONS="$5"
 shift 5
 
-go get k8s.io/code-generator/cmd/{defaulter-gen,conversion-gen,client-gen,lister-gen,informer-gen,deepcopy-gen,openapi-gen}@v0.23.0
+go install k8s.io/code-generator/cmd/{defaulter-gen,conversion-gen,client-gen,lister-gen,informer-gen,deepcopy-gen,openapi-gen}@v0.23.0
```
Comment (Thor-wl) [inline hack/generate-internal-groups.sh:50]: It may be better to include the patch updating `get` to `install` to a spearated commit.

**Comment by Thor-wl** [comment]
BTW, why add `completed` status to `queue`?

File: pkg/apis/scheduling/types.go [hunk 3]
```diff
@@ -51,6 +51,9 @@ const (
 	// PodGroupInqueue means controllers can start to create pods,
 	// is a new state between PodGroupPending and PodGroupRunning
 	PodGroupInqueue PodGroupPhase = "Inqueue"
+
+	// PodGroupCompleted means all the pods of PodGroup are completed
+	PodGroupCompleted PodGroupPhase = "Completed"
```
Comment (k82cn) [inline pkg/apis/scheduling/types.go:56]: How about some Pods are completred but others are failed?

File: pkg/apis/scheduling/v1beta1/types.go (hunk 1 above)
Comment (shinytang6) [reply pkg/apis/scheduling/v1beta1/types.go]: changed it to 7 due to 6 already exists

File: hack/generate-internal-groups.sh (hunk 2 above)
Comment (shinytang6) [reply hack/generate-internal-groups.sh]: ok, remove it

File: pkg/apis/scheduling/types.go (hunk 3 above)
Comment (shinytang6) [reply pkg/apis/scheduling/types.go:56]: My idea is not distinguish succeed & failure pods. In most common scenary, if successful + failed pods == total replicas, we set the PodGroup to Completed.

**Pull request merged**